src/Discord.Net.WebSocket/DiscordSocketClient.cs

```diff
@@ -2243,24 +2243,36 @@ private async Task ProcessMessageAsync(GatewayOpCode opCode, int? seq, string ty
                                         return;
                                     }
 
-                                    var channel = (SocketThreadChannel)guild.GetChannel(data.Id);
+                                    var threadChannel = guild.ThreadChannels.FirstOrDefault(x => x.Id == data.Id);
+                                    var before = threadChannel != null ? threadChannel.Clone() : null;
 
-                                    if (channel == null)
+                                    if (threadChannel != null)
+                                    {
+                                        threadChannel.Update(State, data);
+
+                                        if (data.ThreadMember.IsSpecified)
+                                            threadChannel.AddOrUpdateThreadMember(data.ThreadMember.Value, guild.CurrentUser);
+                                    }
+                                    else
                                     {
+                                        // Thread is updated but was not cached, likely meaning the thread was unarchived.
+                                        // We cache the thread and output an unknown channel warning.
```

Comment:
Is this the right approach? I thought about possibly not emitting the warning and silently not invoking the event.

Reply:
We can use a `Cacheable` type here, since the thread may or may not be provided. Have a look at some of the other events which use the `Cacheable` type

Reply:
That makes sense yeah! I thought about this approach but wasn't sure about changing the event signature. I made the changes, tell me what you think!